wikichron/controls_sidebar_decorator/ControlsSidebar.py

```diff
@@ -0,0 +1,104 @@
+"""
+   ControlsSidebar.py
+
+   Descp: A class to implement the decorator pattern in order to make an
+     easier implementation of the right sidebar
+
+   Created on: 08-02-2019
+
+   Copyright 2019 Youssef 'FRYoussef' El Faqir el Rhazoui <f.r.youssef@hotmail.com>
+"""
+
+import abc
+import dash_html_components as html
+import grasia_dash_components as gdc
+
+
+class ControlsSidebar(metaclass=abc.ABCMeta):
+
+    def __init__(self):
+        self.html_sidebar = []
+        self.stats_section = []
+        self.metrics_section = []
+        self.options_section = []
+
+
+    def fold_button(self):
+        return html.Div(
+            html.Div(
+                id='controls-fold-img-container',
+                className='fold-img-container-cn',
+                children=[
+                    html.P(id='controls-fold-button', className='fold-button-cn')
+                ],
+            ),
+            id='controls-fold-container',
+            style={
+                'display': 'flex'
+            }
+        );
+
+
+    def add_stats_section(self):
+        self.stats_section = html.Div([
+                    html.H5('Network Stats', className='control-title'),
+                    html.Div(id='stats', className='stats-container', children=[])
+                    ], 
+                    className='control-container')
+
+
+    def add_stats_content(self, html_stats):
+        self.stats_section.children[1].children = html_stats
+
+
+    def add_metrics_section(self):
+        self.metrics_section = html.Div([
+                    html.H5('Network Metrics', className='control-title'),
+                    html.Div(children=[])
+                ], className='control-container')
+
+
+    def add_metrics_content(self, html_metrics):
+        self.metrics_section.children[1].children = html_metrics
```

Comment:
Can ever be `add_metrics_content` called separated from `add_metrics_section`? if not, maybe this function should not exist and its functionality be contained inside add_metrics_section.

Reply:
This should be a protected (java) method, I've used it to separete the header section from the content. I think it s a good implementation

Reply:
I mean, do we ever want to draw the header section and not the section content? does it make sense that when we call one of them, we won't immediately call the other one?
By reading the code, my impression is that they always go tightly together, so that it might be non-sense to have them separately and even less to have `content` as part of the ControlsSidebar (Base) interface.
I might be wrong though, so let me know if you still see it unchangeable.